engine/src/platform-internal.h

```diff
@@ -221,6 +224,9 @@ class MCPlatformWindow
 		bool m_use_text_input : 1;
         bool m_is_realized : 1;
 	};
+    
+    // MERG-2015-10-11: [[ DocumentFilename ]] documentFilename property
+    MCStringRef m_document_filename;
```

Comment:
Could you move this to be below m_cursor? The layout of the struct is essentially:

internal stuff
changed flags
property values
object state

m_document_filename counts as a 'property value' :)